mongo/collection.go

```diff
@@ -1791,6 +1787,11 @@ func (coll *Collection) dropEncryptedCollection(ctx context.Context, ef interfac
 	if err := coll.db.Collection(ecocCollection).drop(ctx); err != nil {
 		return err
 	}
+
+	// Drop the data collection.
+	if err := coll.drop(ctx); err != nil {
```

Comment:
Just curious, why did the dropping of the data collection move to be the final operation and not the first?

Reply:
The server expects state collections to be dropped before the data collection. The server warns if the data collection is dropped before state collections per [SERVER-64914](https://jira.mongodb.org/browse/SERVER-64914).